contrib/externs/chrome_extensions.js

```diff
@@ -11214,3 +11214,80 @@ chrome.downloads.DeterminingFilenameEvent.prototype.hasListeners =
  * https://developer.chrome.com/extensions/downloads#event-onDeterminingFilename
  */
 chrome.downloads.onDeterminingFilename;
+
+
+/**
+ * @const
+ * @see https://developer.chrome.com/extensions/declarativeContent
+ */
+chrome.declarativeContent = {};
+
+
+/** @type {string} */
+chrome.declarativeContent.PageStateMatcherInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.ShowPageActionInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.SetIconInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.RequestContentScriptInstanceType;
+
+
+/**
+ * @typedef {{
+ *   pageUrl: (!UrlFilter|undefined),
+ *   css: (!Array<string>|undefined),
+ *   isBookmarked: (boolean|undefined)
+ * }}
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-PageStateMatcher
+ */
+chrome.declarativeContent.PageStateMatcherLiteral;
+
+
+/**
+ * @constructor
+ * @param {?chrome.declarativeContent.PageStateMatcherLiteral=} literalValue
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-PageStateMatcher
+ */
+chrome.declarativeContent.PageStateMatcher = function(literalValue) {};
```

Comment:
Do you expect to write code like this?

```javascript
let matcher = new chrome.declarativeContent.PageStateMatcher(pageStateLiteral);
...
someFunction(matcher.pageUrl); // reference to object property assigned from the literal
```

If so, you need to declare the properties on `PageStateMatcher`.

Reply:
I don't think so, `PageStateMatcher` is only used as a parameter for rules at `chrome.declarativeContent.onPageChanged`.
Check https://developer.chrome.com/extensions/declarativeContent#rules

BTW, you suggested those changes, I first wrote it as a property: https://github.com/google/closure-compiler/pull/2527/commits/c419db9aa7a3c1b6cd91bd9cc9bc0b6354c85ed3#diff-0dbd98fbf87a9a1918041aecf8035ad0R11052

Reply:
I do remember asking you to change that. Thanks for doing so.

I think this is fine then.